### PR TITLE
glib: make sure we use all definitions returned by pkg-config

### LINKED ISSUE #718: glib bindings: linking fails on OS X, FreeBSD
On both systems gobject-introspection is successfully found
```
-- Checking for module 'gobject-2.0>=2.36'
--   Found gobject-2.0, version 2.46.2
-- Include Binding glib
-- Checking for module 'gobject-introspection-1.0>=1.38'
--   Found gobject-introspection-1.0, version 1.46.0
```

Fails on both systems very similarly.
```
[ 72%] Linking C shared library ../../../lib/libgelektra-4.0.dylib
cd /Users/mpranj/workspace/libelektra/build/src/bindings/glib && /usr/local/Cellar/cmake/3.5.2/bin/cmake -E cmake_link_script CMakeFiles/gelektra-4.0.dir/link.txt --verbose=1
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc   -std=gnu99  -Wno-deprecated-declarations  -Wno-long-long -pedantic -Wno-variadic-macros -Wall -Wextra -Wno-overlength-strings -Wsign-compare -Wfloat-equal -Wformat-security -Wshadow -Wcomments -Wtrigraphs -Wundef -Wuninitialized -Winit-self -Wsign-compare -Wfloat-equal -Wformat-security -O2 -g -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -dynamiclib -Wl,-headerpad_max_install_names  -o ../../../lib/libgelektra-4.0.dylib -install_name @rpath/libgelektra-4.0.dylib CMakeFiles/gelektra-4.0.dir/gelektra-kdb.c.o CMakeFiles/gelektra-4.0.dir/gelektra-key.c.o CMakeFiles/gelektra-4.0.dir/gelektra-keyset.c.o ../../../lib/libelektra-kdb.0.8.16.dylib -lgobject-2.0 -lglib-2.0 -lintl ../../../lib/libelektra-core.0.8.16.dylib -Wl,-rpath,/Users/mpranj/workspace/libelektra/build/lib
ld: library not found for -lgobject-2.0
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/libgelektra-4.0.dylib] Error 1
```

### PR DESCRIPTION
**Pull request by manuelm**
Fixes #718

### CODE REVIEW

**Comment by manuelm** [comment]
jenkins build bindings please

File: src/bindings/glib/tests/CMakeLists.txt [hunk 1]
```diff
@@ -14,9 +12,7 @@ macro (do_glib_test file)
 		COMPILE_DEFINITIONS HAVE_KDBCONFIG_H
 	)
 
-	set (GLIBS ${GELEKTRA_LIBRARY} ${GOBJECT_LIBRARIES})
-	tool_link_elektra (${name})
-	target_link_libraries (${name} ${GLIBS})
+	target_link_libraries (${name} PRIVATE ${GELEKTRA_LIBRARY})
```
Comment (markus2330) [inline src/bindings/glib/tests/CMakeLists.txt]: does this needs 2.8.12, too?

Comment (manuelm) [inline src/bindings/glib/tests/CMakeLists.txt]: why are you asking?

Comment (markus2330) [inline src/bindings/glib/tests/CMakeLists.txt]: only to understand which parts require the new CMake, it is not documented above.

Comment (manuelm) [inline src/bindings/glib/tests/CMakeLists.txt]: The changelog says PUBLIC | PRIVATE | INTERFACE has been introduced in 2.8.12, so yeah.
https://cmake.org/cmake/help/v3.0/policy/CMP0023.html

Comment (manuelm) [inline src/bindings/glib/tests/CMakeLists.txt]: btw target_link_libraries accepts both, libraries and linker flags (including directories without -L)

**Pull request merged**